BrainPortal/lib/boutiques_input_value_fixer.rb

```diff
@@ -0,0 +1,182 @@
+          block_inputs(descriptor_dup, members - fixation.keys)
+          # a better solution is to delete rest of group params completely
+          # a bit more complex though and might result in recursive code or nested loops
+        end
+      end
+
+      # all-or-none is not reflected in dynamic gui, uncomment once fixed
+
+      # removes  'one-is-required' or disables group when one or more element fixed, e.g.
+      # if g.all_or_none && members.length != g.members.length
+      #   # if (g.members & skipped).present?
+      #   #   # todo delete all member inputs, or disable by injecting pairwise required/disable dependencies
+      #   # end
+      #   if (fixation.keys & g.members - skipped).present? # if one is set, rest should be to
+      #     g.members.each do |i_id|
+      #       begin
+      #         input = descriptor.input_by_id(i_id)
+      #       rescue CbrainError  # if descriptor was already processed
+      #         next
+      #       end
+      #       input.optional = false
+      #     end
+      #   end
+      # end
+
+      # presently one-is-required is checked only statically, no GUI support
+      # removes one-is-required flag if one element fixed
+      # if g.one_is_required && members.length != g.members.length
+      #   if (fixation.keys & g.members - skipped).present?
+      #     g.one_is_required == false
+      #   end
+      # end
+
+      g.members = members
+    end
+    descriptor_dup.groups = descriptor_dup.groups.select {|g| g.members.present? } # delete empty group
+
+    # delete fixed inputs
+    descriptor_dup.inputs = descriptor_dup.inputs.select { |i| ! fixation.key?(i.id) } # filter out fixed inputs
+
+    # crude erase of fixed inputs from dependencies.
+    #
+    descriptor_dup.inputs.each do |i|
+      i.requires_inputs = i.requires_inputs - fixation.keys if i.requires_inputs.present?
+      i.disables_inputs = i.disables_inputs - fixation.keys if i.disables_inputs.present?
+      i.value_requires.each { |v, a| i.value_requires[v] -= fixation.keys } if i.value_requires.present?
+      i.value_disables.each { |v, a| i.value_disables[v] -= fixation.keys } if i.value_disables.present?
+    end
+
+    descriptor_dup
+  end
+
+  # this is blocks an input parameter, rather than explicitely deleting it
+  # it is a bit unconvential yet expected to be used for relatively rare case when fixing or deleting
+  # one input has implications.
+  # Assuming the the boutiques developer(s) test their results
+  def block_inputs(descriptor, input_ids)
+    input_ids.each do |input_id|
+
+      input = descriptor.input_by_id(input_id) rescue next
+      #input.disables_if input.disables_inputs.present?
+      input.disables_inputs ||= []
+      input.disables_inputs |= [input_id]
+      input.name += " --- disabled by admin ---"
+    end
+  end
+
+  # adjust descriptor to allow check # of supplied files
+  def descriptor_for_before_form
+    delete_fixed_inputs(super)
```

Comment:
I am not sure this is necessary. I don't really see why someone configuring a custom section of a desccriptor woudl provide a fixed value for a File input. Can you think of a case like that?

Reply:
atlas? config file?